### DISCUSSION TITLE
How to create iOS library

### DISCUSSION THREAD

**Discussion by ienground** [sunny-chung/kdatetime-multiplatform#4] [Q&A] [answered]
```
you first need to have a copy of Android SDK. Create a local.properties under the root directory of the repository, then set sdk.dir=/path/to/android-sdk in this file. Then, run gradle assemble. If it succeeds, you will find the aar under the directory ./build/outputs/aar/.
```

As you mentioned, I created AAR file to use new version of this library. However, it seems not to use on iOS... Should I create another library file? Then.. how can we create the library file of iOS? 
Thank you!

```
gradle build
```
<img width="363" alt="스크린샷 2024-12-03 오후 1 40 29" src="https://github.com/user-attachments/assets/d999815e-b6ed-49b7-bd07-236d842a3c47">
I cannot use this file

**Comment by sunny-chung** [comment]
The artifacts of KMP iOS libraries are `.klib` and `metadata.jar`, but it seems quite hard to import these files directly into KMP projects. The simplest way would probably be installing the KMP library into the local maven repository, then make use of this local repository in KMP application projects.

To install the library to the local maven repository, run `gradle publishToMavenLocal` in KDateTime. Remember to rename the version before execution in order to differentiate the one just built from public ones.

In KMP application projects, update the `build.gradle.kts` file:
```kotlin
repositories {
    // ...
    mavenLocal()
}

kotlin {
    sourceSets {
        val commonMain by getting {
            dependencies {
                api("io.github.sunny-chung:kdatetime-multiplatform:<version>")
                // ...
            }
        }
        // ...
```

**Reply by ienground** [reply]
```
* What went wrong:
Execution failed for task ':signAndroidReleasePublication'.
> Cannot perform signing task ':signAndroidReleasePublication' because it has no configured signatory
```
this happened to me ... :'(

**Reply by ienground** [reply]
Commenting out this part solved it!
in `sunnychung.publication.gradle.kts` in `convention-plugins/src/main/kotlin`
```
// Signing artifacts. Signing.* extra properties values will be used
//signing {
//    sign(publishing.publications)
//}
```

Thanks for your help :>

**Reply by sunny-chung** [reply, marked as answer]
Glad that your issue has been solved. I may also update the script not to enable signing if it is to publish to local maven.
